doc/release/upcoming_changes/25080.change.rst

```diff
@@ -0,0 +1,2 @@
+* Redefine `numpy.bool` as an alias for `numpy.bool_` (as opposed to the `bool`
+it was until NumPy 1.24) for Array API compatibility.
```

Comment:
Maybe also make a separate release note for the repr of `np.True_` and `np.False_`, mentioning that those are still available but the canonical names are now `np.bool(True)` and `np.bool(False)`.

Reply:
Aren't those canonical names slower?

Reply:
Resolving as we decided to stay with `np.True_` and `np.False_` for reprs.